edurange_refactored/flask/modules/routes/student_routes.py

```diff
@@ -84,6 +102,94 @@ def jwt_test():
         'user_id' : current_user_id,
         'user_role': current_user_role
     })
+
+@blueprint_edurange3_student.route('/groups', methods=['GET']) # DEV_ONLY
+@jwt_and_csrf_required
+def get_groups():
+    db_ses = db.session 
+    all_groups = db_ses.query(StudentGroups).all()
+    all_groups_list = []
+
+    for group in all_groups:
+        group_info = {
+            "id": group.id,
+            "name": group.name,
+            "owner_id": group.owner_id,
+            "code": group.code,
+            "hidden": group.hidden,
+        }
+    print(jsonify(all_groups_list))
+    print(all_groups_list)
+    all_groups_list.append(group_info)
+    return jsonify(all_groups_list)
+    
+
+@blueprint_edurange3_student.route("/make_scenario", methods=["POST"])
+@jwt_and_csrf_required
+def make_scenario():
+    
+    user_is_admin_and_instructor()
+
+    db_ses = db.session
+    name = request.form.get("scenario_name")
+    '''
+
+    s_type = identify_type(request.form)
+    own_id = session.get("_user_id")
+        group = request.form.get("scenario_group")
+
+        students = (
+            db_ses.query(User.username)
+            .filter(StudentGroups.name == group)
+            .filter(StudentGroups.id == GroupUsers.group_id)
+            .filter(GroupUsers.user_id == User.id)
+            .all()
+        )
+
+        Scenarios.create(name=name, description=s_type, owner_id=own_id) #creates database entry
+        NotifyCapture(f"Scenario {name} has been created.")
+        #Notification.create(details=something, date=something)
+        s_id = db_ses.query(Scenarios.id).filter(Scenarios.name == name).first()
+
+        s_id_list = list(s_id._asdict().values())[0]
+
+        scenario = Scenarios.query.filter_by(id=s_id_list).first()
+        scenario.update(status=7)
+        g_id = db_ses.query(StudentGroups.id).filter(StudentGroups.name == group).first()
+        g_id = g_id._asdict()
+
+        # JUSTIFICATION:
+        # Above queries return sqlalchemy collections.result objects
+        # _asdict() method is needed in case celery serializer fails
+        # Unknown exactly when this may occur, maybe version differences between Mac/Linux
+
+        for i, s in enumerate(students):
+            students[i] = s._asdict()
+
+        # s_id, g_id = s_id._asdict(), g_id._asdict()
+
+        gid = g_id["id"]
+        student_ids = db_ses\
+                    .query(GroupUsers.id)\
+                    .filter(GroupUsers.group_id == gid)\
+                    .all()
+
+        namedict = gen_chat_names(student_ids, s_id_list)
```

Comment:
Does this commented out code plan to be used later? I also see an assignment to a variable called namedict inside the commented out code, not sure if its related to what Jack mentioned.